core/src/main/java/com/linecorp/armeria/client/Endpoint.java

```diff
@@ -433,6 +433,32 @@ public String authority() {
         return this.authority = authority;
     }
 
+    /**
+     * Converts this endpoint into a URI string.
+     *
+     * @return the URI string
+     */
+    public String toURI() {
```

Comment:
How about using `authority()` method? If you do that, you do not have to consider when the `endpoint` is a group.

Reply:
Current `authority()` return `domain name` when `domain name` and `ipAddr` are set.
But what I expect to `toURI()` is return `ipAddr` under that condition.

Reply:
When an `Endpoint` has a domain name and IP address, we should use 
IP address + port to connect to the remote host and the domain name as the authority header. So if we just use `ipAddr` the domain name vanishes. 
@Trustin and I have talked about this and he said it's probably better to add `newClient(String scheme, Endpoint endpoint, ClientOptions)` to `ClientFactory` so that we can still use the value.

Reply:
Great! I understand.
I'll update it. Thanks!